ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/Send.hs

```diff
@@ -196,26 +272,26 @@ runSend uiWalletFace sendInputs = do
     QDialog.Rejected -> SendCancel
 
 fillSendOptions :: Send -> IO (Validation Errors SendOptions)
-fillSendOptions Send{..} = do
-  amount <- fromString <$> QLineEdit.text amountEdit
-  address <- fromString <$> QLineEdit.text addressEdit
-
+fillSendOptions s@Send{..} = do
+  enabledReceivers <- getEnabledReceivers s
+  amounts <- mapM (\r'@Receiver{..} -> liftM2 (,) (return $ r') (fromString <$> QLineEdit.text amountEdit)) enabledReceivers
+  addresses <- mapM (\r'@Receiver{..} -> liftM2 (,) (return $ r') (fromString <$> QLineEdit.text addressEdit)) enabledReceivers
   accounts <- case accountSelector of
     Left aIdx -> return [aIdx]
     Right AccountSelector{..} -> map aIdx <$> filterM (QAbstractButton.isChecked . checkbox) checkboxes
 
   let UiWalletFace{..} = uiWalletFace
-
   let
-    vAddress = validate [AddressEmpty] (not . null) address `bindValidation` \a ->
+    vAddresses = traverse (\(r', address) -> validate [(r', AddressEmpty)] (not . null) address `bindValidation` \a ->
```

Comment:
Please check if there's `flip traverse` somewhere in Prelude and use that, I believe that'd be cleare. Also, this whole block looks like a mess, I'm really sad about `Data.Validation` :/